lib/get-config.js

```diff
@@ -4,9 +4,25 @@ var path = require('path');
 var assign = require('lodash').assign;
 var existsSync = require('exists-sync');
 var chalk = require('chalk');
-var rules = require('./rules');
+var defaultRules = require('./rules');
+
+var KNOWN_ROOT_PROPERTIES = ['extends', 'rules', 'pending', 'ignore', 'plugins'];
+var KNOWN_EXTENDED_CONFIGURATION = ['recommended'];
```

Comment:
Instead of this, lets create a `lib/config/index.js` with:

```js
"use strict";

module.exports = {
  'recommended': require('./recommended')
};
```

This makes it much more similar to how `rules` are done, and how plugins will work...

Then we can use `name in KNOWN_CONFIGURATIONS` below in `_getExtendedRules`...